7b_temp_merge/src/merge_all_temp_dat.R

```diff
@@ -0,0 +1,117 @@
+merge_temp_data <- function(outind, wqp_ind, coop_ind) {
+
+  outfile <- as_data_file(outind)
+
+  wqp_in <- sc_retrieve(wqp_ind)
+  wqp_dat <- feather::read_feather(wqp_in)
+
+  coop_in <- sc_retrieve(coop_ind)
+  coop_dat <- feather::read_feather(coop_in)
+
+  total_obs <- nrow(wqp_dat) + nrow(coop_dat)
+
+  # merge and remove duplicate observations - defined by rounding to 1 decimal point for depth and temp
+  all_dat <- select(wqp_dat, date = Date, time, timezone, depth, temp = wtemp, nhd_id = id, source_id = wqx.id, source_site_id = wqx.id) %>%
+    mutate(source = 'wqp') %>%
+    bind_rows(select(coop_dat, date = DateTime, time, timezone, depth, temp, nhd_id = site_id, source_id = state_id, source_site_id = site, source)) %>%
+    mutate(depth = round(depth, 2),
+           depth1 = round(depth, 1),
+           temp = round(temp, 2),
+           temp1 = round(temp, 1)) %>%
+    distinct()
+
+  all_dat_distinct <- all_dat %>%
+    mutate(timezone = ifelse(is.na(time), NA, timezone)) %>%
+    distinct(nhd_id, date, time, depth1, temp1, .keep_all = TRUE) %>%
+    select(-depth1, -temp1)
+
+  # note - could still have lake/date/depth duplicates here
+  # the risk of removing time from the above distinct() is that if you have
+  # multiple observations per day, you could inadvertently lose obs that are distinct
+  # in time but happen to have the same values (e.g., noon and 1pm surface samples that measure
+  # the same temperature). The risk of not removing them is that you have multiple data sources
+  # repeating the same data - but may not have a time stamp, for example. Will keep for now, sort out
+  # in next step where we resolve time.
+
+  cat(total_obs - nrow(all_dat_distinct), 'temperature observations were duplicated across WQP and cooperator data files and removed.')
+
+  # get rid of egregious values before posting data or choosing time stamps
+  all_dat_distinct_noout <- mutate(all_dat_distinct, month = lubridate::month(date)) %>%
+    filter(!(month %in% c(1, 2) & temp > 10)) %>%
+    filter(!(month %in% c(7, 8) & depth < 0.5 & temp < 10))
```

Comment:
These values seem reasonable but I wonder if we could make a function that is based on latitude and uses long-term water temp averages for each month or something like that. Would be useful for munging continental water temp data where Jan/Feb temp outliers will be much different for different areas of the country

Reply:
Ah yup that makes more sense to have in the reality check portion of the pipeline. I also forgot to mention that I think the way you have it is good and that making a function would be overkill for this pipeline, but in the future it might be useful